Mf2/Parser.php

```diff
@@ -949,25 +1012,37 @@ public function parseH(\DOMElement $e, $is_backcompat = false) {
 			$this->elementPrefixParsed($u, 'u');
 		}
 
+		$temp_dates = array();
+
 		// Handle dt-*
 		foreach ($this->xpath->query('.//*[contains(concat(" ", @class), " dt-")]', $e) as $dt) {
 			if ($this->isElementParsed($dt, 'dt')) {
 				continue;
 			}
 
-			$dtValue = $this->parseDT($dt, $dates);
+			$dtValue = $this->parseDT($dt, $dates, $impliedTimezone);
 
 			if ($dtValue) {
 				// Add the value to the array for dt- properties
 				foreach (mfNamesFromElement($dt, 'dt-') as $propName) {
-					$return[$propName][] = $dtValue;
+					$temp_dates[$propName][] = $dtValue;
 				}
 			}
-
 			// Make sure this sub-mf won’t get parsed as a top level mf
 			$this->elementPrefixParsed($dt, 'dt');
 		}
 
+		foreach ($temp_dates as $propName => $data) {
+			foreach ( $data as $dtValue ) {
+
+				if ( $impliedTimezone && preg_match('/[+|-]\d{2}\d{2}/i', $dtValue, $matches) == 0 ) {
```

Comment:
This is after the new method `normalizeTimezoneOffset()` is called, which removes the colon. This isn't explicitly stated in the spec, but I inferred it from: (bold is my emphasis)

> However the colons ":" separating the hours and minutes of any timezone offset are optional **and discouraged** in order to make it less likely that a timezone offset will be confused for a time.